components/widgets/climate/carbon-flux/selectors.js

```diff
@@ -0,0 +1,273 @@
+import { createSelector, createStructuredSelector } from 'reselect';
+import { format } from 'd3-format';
+import { formatNumber } from 'utils/format';
+import isEmpty from 'lodash/isEmpty';
+// get list data
+const getData = (state) => state.data;
+const getSettings = (state) => state.settings;
+const getColors = (state) => state.colors;
+const getIndicator = (state) => state.indicator;
+const getLocationName = (state) => state.locationLabel;
+const getSentences = (state) => state.sentences;
+const getAdminLevel = (state) => state.adminLevel;
+
+export const parseData = createSelector(
+  [getData, getSettings],
+  (totalData, settings) => {
+    if (!totalData) return null;
+    const { startYear, endYear } = settings;
+
+    // TODO: return as object
+    return [
+      totalData.length &&
+        Object.keys(totalData[0]).reduce((result, k) => {
+          result[k] = totalData[0][k] / (endYear - startYear + 1);
+          return result;
+        }, {}),
+    ];
+  }
+);
+
+export const parseConfig = createSelector(
+  [parseData, getColors],
+  (data, colors) => {
+    if (!data || isEmpty(data)) return null;
+    const {
+      carbonFlux: { emissions, netEmissions, removals, netRemovals },
+    } = colors;
+
+    const maxValue = Math.ceil(
+      Math.abs(
+        Object.values(data[0]).reduce((a, b) => (data[a] > data[b] ? b : a))
+      ) * 1.2
+    );
+
+    const maxValueDigits = maxValue.toString().length - 2;
+
+    // format numbers operate with array
+    const tickFormatter = (value) =>
+      format('.2r')(value * `1e-${maxValueDigits}`);
+
+    // make ticks multiple of 0.5
+    const tick = Math.ceil(tickFormatter(maxValue) / 0.5) * 0.5;
+
+    // calculate ticks array
+    const range = (start, stop, step) =>
+      Array.from(
+        { length: (stop - start) / step + 1 },
+        (_, i) => start + i * step
+      );
+    const ticks = Array.from(range(-tick, tick, 0.5)).map(
+      (t) => t * `1e+${maxValueDigits}`
+    );
```

Comment:
Nice solution! I guess since we're using the maxValueDigits to get the order of magnitude, we should do the same for the `xAxis.label.value` too?

i.e. replace G with M, k etc? 🤔  but that might be a pain. Alternatively we could use scientific notion in the tick labels (G,M,k etc) and make the axis label simple `tCO2e/year`